### PR TITLE
More information in exception when exec returns an error code

### PR DESCRIPTION
**Pull request by thinkbeforecoding**
This one is here to help investigate problems when the git exe fails.
We have a randomly recurring 128 error code, and no more info in CI builds.
Having more info could help.

(this one is courtesy of Clem.. if you can un-ban him ?..)

### CODE REVIEW

**Comment by pchalamet** [comment]
:white_check_mark: [Build full-build 4.3.36 completed](https://ci.appveyor.com/project/full-build/full-build-shu78/build/4.3.36) (commit https://github.com/full-build/full-build/commit/03364e93c7 by @)

File: src/FullBuild/IO.fs [hunk 1]
```diff
@@ -14,14 +14,23 @@
...
       Info : string
       Out: string list
       Error: string list }
 
 let private resultToError execResult =
-    if execResult.Code <> 0 then (sprintf "Operation '%s' failed with error %d" execResult.Info execResult.Code) |> Some
+    if execResult.Code <> 0 then 
+        let prefix p = function [] -> [] | l -> p :: l
+
+        sprintf "Operation '%s' failed with error %d" execResult.Info execResult.Code
```
Comment (pchalamet) [inline src/FullBuild/IO.fs]: Would be better to always write out and error.
Just create a concatenated list of strings and get rid of prefix is not that useful.

File: src/FullBuild/IO.fs [hunk 2]
```diff
@@ -14,14 +14,23 @@
...
 
 let private resultToError execResult =
-    if execResult.Code <> 0 then (sprintf "Operation '%s' failed with error %d" execResult.Info execResult.Code) |> Some
+    if execResult.Code <> 0 then 
+        let prefix p = function [] -> [] | l -> p :: l
+
+        sprintf "Operation '%s' failed with error %d" execResult.Info execResult.Code 
+        :: prefix "Out:" execResult.Out
+        @ prefix "Error:" execResult.Error
+        |> String.concat Environment.NewLine
```
Comment (pchalamet) [inline src/FullBuild/IO.fs]: |> must be indented

File: src/FullBuild/IO.fs (hunk 1 above)
Comment (thinkbeforecoding) [reply src/FullBuild/IO.fs]: It just add the prefix to show which part of the text is what. And just doesn't bloat output when there is no Out or Error.

File: src/FullBuild/IO.fs (hunk 2 above)
Comment (thinkbeforecoding) [reply src/FullBuild/IO.fs]: This is standard indentation for |>...

**Comment by thinkbeforecoding** [comment]
This PR is intended to help in #194

File: src/FullBuild/IO.fs (hunk 2 above)
Comment (sergii-s) [reply src/FullBuild/IO.fs]: For me it's the same as everywhere in the full-build, so ok

Comment (pchalamet) [reply src/FullBuild/IO.fs]: Yes it's strange and indentation is still not ok.

File: src/FullBuild/IO.fs (hunk 1 above)
Comment (pchalamet) [reply src/FullBuild/IO.fs]: prefix is just unreadable - get rid of this please and just ensure output is deterministic.

File: src/FullBuild/IO.fs (hunk 2 above)
Comment (thinkbeforecoding) [reply src/FullBuild/IO.fs]: It's compiling and aligned on previous lines, I don't see what you expect ?

File: src/FullBuild/IO.fs (hunk 1 above)
Comment (thinkbeforecoding) [reply src/FullBuild/IO.fs]: Removed

**Comment by pchalamet** [comment]
:white_check_mark: [Build full-build 4.3.37 completed](https://ci.appveyor.com/project/full-build/full-build-shu78/build/4.3.37) (commit https://github.com/full-build/full-build/commit/2e42da756b by @)

**Pull request merged**